spark-connector/src/main/java/com/datastrato/gravitino/spark/table/SparkBaseTable.java

```diff
@@ -0,0 +1,129 @@
+import org.apache.spark.sql.connector.catalog.SupportsWrite;
+import org.apache.spark.sql.connector.catalog.Table;
+import org.apache.spark.sql.connector.catalog.TableCapability;
+import org.apache.spark.sql.connector.catalog.TableCatalog;
+import org.apache.spark.sql.connector.read.ScanBuilder;
+import org.apache.spark.sql.connector.write.LogicalWriteInfo;
+import org.apache.spark.sql.connector.write.WriteBuilder;
+import org.apache.spark.sql.types.Metadata;
+import org.apache.spark.sql.types.MetadataBuilder;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+import org.apache.spark.sql.types.StructType$;
+import org.apache.spark.sql.util.CaseInsensitiveStringMap;
+
+/**
+ * Provides schema info from Gravitino, IO from the internal spark table. The specific catalog table
+ * could implement more capabilities like SupportsPartitionManagement for Hive table, SupportsIndex
+ * for JDBC table, SupportsRowLevelOperations for Iceberg table.
+ */
+public class SparkBaseTable implements Table, SupportsRead, SupportsWrite {
+  private Identifier identifier;
+  private com.datastrato.gravitino.rel.Table gravitinoTable;
+  private TableCatalog sparkCatalog;
+  private Table lazySparkTable;
+  private PropertiesConverter propertiesConverter;
+
+  public SparkBaseTable(
+      Identifier identifier,
+      com.datastrato.gravitino.rel.Table gravitinoTable,
+      TableCatalog sparkCatalog,
+      PropertiesConverter propertiesConverter) {
+    this.identifier = identifier;
+    this.gravitinoTable = gravitinoTable;
+    this.sparkCatalog = sparkCatalog;
+    this.propertiesConverter = propertiesConverter;
+  }
+
+  protected Table getSparkTable() {
+    if (lazySparkTable == null) {
+      try {
+        lazySparkTable = this.sparkCatalog.loadTable(identifier);
+      } catch (NoSuchTableException e) {
+        throw new RuntimeException(e);
+      }
+    }
+    return lazySparkTable;
+  }
+
+  @Override
+  public String name() {
+    return identifier.toString();
+  }
+
+  @Override
+  @SuppressWarnings("deprecation")
+  public StructType schema() {
+    List<StructField> structs =
+        Arrays.stream(gravitinoTable.columns())
+            .map(
+                column -> {
+                  String comment = column.comment();
+                  Metadata metadata = Metadata.empty();
+                  if (comment != null) {
+                    metadata =
+                        new MetadataBuilder()
+                            .putString(ConnectorConstants.COMMENT, comment)
+                            .build();
+                  }
+                  return StructField.apply(
+                      column.name(),
+                      SparkTypeConverter.toSparkType(column.dataType()),
+                      column.nullable(),
+                      metadata);
+                })
+            .collect(Collectors.toList());
+    return StructType$.MODULE$.apply(structs);
```

Comment:
Do yo support nest fields here?

Reply:
no, only support Integer, bool, String, more types will be supported in another PR